src/observers/link_prefetch_observer.js

```diff
@@ -9,10 +9,14 @@ import { StreamMessage } from "../core/streams/stream_message"
 import { FetchMethod, FetchRequest } from "../http/fetch_request"
 import { prefetchCache, cacheTtl } from "../core/drive/prefetch_cache"
 
+const observedEvents = {
+  "mouseenter": "mouseleave",
+  "touchstart": "touchcancel"
```

Comment:
@seanpdoyle I don't think `touchcancel` is mimicking  `mouseleave` here. `touchcancel` means that a touch point has been interrupted in an unexpected manner, but a touch point could leave the target in another ways.

Reply:
I think the spirit of the teardown-side of event listening is that unless the click is navigated to `100ms` after the setup-side, the prefetching should be reset. 

Would [touchend](https://developer.mozilla.org/en-US/docs/Web/API/Element/touchend_event) be more appropriate? Would mapping `observedEvents` to an array of listeners be better?

```js
const observedEvents = {
  "mouseenter": ["mouseleave"],
  "touchstart": ["touchcancel", "touchend"],
  "focusin": ["focusout"]
}
```

Reply:
@seanpdoyle what do you think about 938b355733a075ddb3ac20721ba45b3af429f9ad ? It's basically the same idea, but we check if the prefetch request is still relevant on `touchmove` and `touchend` events, instead.

Reply:
@afcapel I like that approach. I wonder if there's a way to make the event listener callback generic enough to apply to all types.

Maybe something like:

```js
#checkIfPrefetchValid(event) {
  const targets = "targetTouches" in event ?
    [event.target, ...event.targetTouches] :
    [event.target]
  const interactingWithPrefetchedLink = targets.some((target) => target === this.#prefetchedLink)

  if (interactingWithPrefetchedLink) {
    // .. do the cancellation
  }
}
```

That change could support the idea of `observedEvents` mapping from "start" to "end" events in a generic enough way to support expanding support for incorporate focus change for keyboard navigators.

Reply:
> Regardless of focus support, I think an abstraction that correlates which events start a prefetch with the events that cancel a prefetch has value.

Fair enough, If you can update the PR I'll merge it 👍  A test exercising the prefetch cancellation would also be useful.

Regarding prefetching on `focusin`/`focusout`, as @brunoprietog [pointed](https://github.com/hotwired/turbo/pull/1101#discussion_r1458077922) the keyboard navigation may need a special timing. But honestly, I'm not sure, we'd have to test it on realistic conditions to tell.

Reply:
> A test exercising the prefetch cancellation would also be useful

Unfortunately, I can't find a way to start then end a `touch` event in Playwright. The closest that I came to finding a solution was the [Locator.tap](https://playwright.dev/docs/api/class-locator#locator-tap) method, but that ends up definitively tapping the link as a navigation, even when immediately followed by tapping elsewhere in the screen.